src/System.Net.NetworkInformation/src/System/Net/NetworkInformation/NetworkAddressChange.OSX.cs

```diff
@@ -61,10 +67,39 @@ public class NetworkChange
             {
                 lock (s_lockObj)
                 {
-                    bool hadSubscribers = s_addressChangedSubscribers != null;
+                    bool hadSubscribers = s_addressChangedSubscribers != null || s_availabilityChangedSubscribers != null;
                     s_addressChangedSubscribers -= value;
 
-                    if (hadSubscribers && s_addressChangedSubscribers == null)
+                    if (hadSubscribers && s_addressChangedSubscribers == null && s_availabilityChangedSubscribers == null)
```

Comment:
Do we need to check s_availabilityChangedSubscribers twice?  Seems like it could be:
```C#
bool hadAddressChangedSubscribers = s_addressChangedSubscribers != null;
s_addressChangedSubscribers -= value;
if (hadAddressChangedSubscribers && s_addressChangedSubscribers == null && s_availabilityChangedSubscribers == null)
{
    StopRunLoop();
}
```
or potentially we could just always call StopRunLoop if they're both null, regardless of what they were before, and just have StopRunLoop be a nop if nothing's currently running.